app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -118,11 +122,19 @@ class ConversationMessagesViewModel @Inject constructor(
     }
 
     private fun loadPaginatedMessages() = viewModelScope.launch {
-        val paginatedMessagesFlow = getMessageForConversation(conversationId)
+        val lastReadIndex = when (val result = getConversationUnreadEventsCount(conversationId)) {
+            is GetConversationUnreadEventsCountUseCase.Result.Success -> result.amount.toInt()
+            is GetConversationUnreadEventsCountUseCase.Result.Failure -> 0
+        }
+
+        val paginatedMessagesFlow = getMessageForConversation(conversationId, lastReadIndex)
             .flowOn(dispatchers.io())
             .cachedIn(this)
 
-        conversationViewState = conversationViewState.copy(messages = paginatedMessagesFlow)
+        conversationViewState = conversationViewState.copy(
+            messages = paginatedMessagesFlow,
+            firstuUnreadEventIndex = max(lastReadIndex - 1, 0)
```

Comment:
does this makes sense ? I think what you tried to do here is to make sure we do not get negative values ? if yes I think abs() will make more sense, if not then I do not see the use case for this 🤔